src/Provider/Node/FallbackNodeProvider.php

```diff
@@ -42,12 +42,12 @@ public function __construct(array $providers)
      * Returns the system node ID by iterating over an array of node providers
      * and returning the first non-empty value found
      *
-     * @return string System node ID as a hexadecimal string
+     * @return string|null System node ID as a hexadecimal string
      */
     public function getNode()
     {
         foreach ($this->nodeProviders as $provider) {
-            if ($node = $provider->getNode()) {
+            if ((bool) ($node = $provider->getNode())) {
```

Comment:
What about something like this:
```php
$node = $provider->getNode();
if ($node !== null) { 
    return $node;
}
```

When PHPStan warns you, that there is implicit boolean conversion in condition, it does not mean that it should be replaced with explicit 😄